runtime/src/chpl-gpu.c

```diff
@@ -257,8 +257,10 @@ typedef struct kernel_cfg_s {
   void*** kernel_params;
   int n_host_registered;
   int cur_host_registered_var;
-  void** host_registered_var_boxes;
-  void*** host_registered_vars;
+  void** host_registered_var_boxes; // array of device pointers
+  void*** host_registered_vars;     // array of pointers to device pointer
+                                    // (stored in host_registered_var_boxes)
+  void** host_registered_vars_host_ptrs; // array of host pointers
```

Comment:
I am slowly getting a negative feeling for the long variable names here :D I'm OK for now, but I may not be for the next PR :)

Reply:
I_blame_snake_case.